src/Components/Table/CustomFooter.tsx

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import {
+  useGridApiContext,
+  useGridSelector,
+  gridPaginationModelSelector,
+  gridPageCountSelector,
+  gridPageSizeSelector,
+} from "@mui/x-data-grid";
+import {
+  Box,
+  FormControl,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  Typography,
+  Pagination,
+} from "@mui/material";
+
+function CustomPagination() {
+  const apiRef = useGridApiContext();
+  const paginationModel = useGridSelector(apiRef, gridPaginationModelSelector);
+  const pageCount = useGridSelector(apiRef, gridPageCountSelector);
+  const pageSize = useGridSelector(apiRef, gridPageSizeSelector);
+  const rowsCount = apiRef.current.getRowsCount();
+
+  const start = paginationModel.page * pageSize + 1;
+  const end = Math.min((paginationModel.page + 1) * pageSize, rowsCount);
+
+  const defaultPageSize = 10;
+
+  const generatePageSizeOptions = (count: number) => {
+    const options = [10, 25, 50];
+    const maxOption = Math.min(count, 100);
+    if (!options.includes(maxOption)) {
+      options.push(maxOption);
+    }
+    if (!options.includes(defaultPageSize)) {
+      options.push(defaultPageSize);
+    }
+    return Array.from(new Set(options)).sort((a, b) => a - b);
+  };
+
+  const pageSizeOptions = generatePageSizeOptions(rowsCount);
+
+  const handlePageSizeChange = (event: SelectChangeEvent<number>) => {
+    const value = event.target.value as number;
+    apiRef.current.setPageSize(value);
+  };
+
+  React.useEffect(() => {
```

Comment:
change the import to `import {useEffect} from 'react'` and then use `useEffect( () => {},[])` instead of `React.useEffect( () => {},[])`

Reply:
done